benchmark/cajita/Cajita_HaloPerformance.cpp

```diff
@@ -0,0 +1,274 @@
+/****************************************************************************
+ * Copyright (c) 2018-2022 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "../Cabana_BenchmarkUtils.hpp"
+
+#include <Cabana_Core.hpp>
+#include <Cajita.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+#include <ratio>
+#include <string>
+#include <vector>
+
+#include <mpi.h>
+
+//---------------------------------------------------------------------------//
+// Performance test.
+template <class Device>
+void performanceTest( std::ostream& stream,
+                      const std::vector<int> problem_global_cells_per_dim,
+                      const std::string& test_prefix,
+                      std::vector<int> problem_halo_width, MPI_Comm comm )
+{
+
+    using exec_space = typename Device::execution_space;
+    using device_type = typename exec_space::device_type;
+
+    // the size of halo_width
+    int halo_width_size = problem_halo_width.size();
+
+    // the size of problem
+    int global_cells_per_dim_size = problem_global_cells_per_dim.size();
+
+    // number of runs
+    int num_run = 10;
+
+    for ( int n = 0; n < global_cells_per_dim_size; ++n )
+    {
+
+        int comm_size;
+        MPI_Comm_size( comm, &comm_size );
+
+        // Get Cartesian comm
+        std::array<int, 3> ranks_per_dim;
+        for ( std::size_t d = 0; d < 3; ++d )
+            ranks_per_dim[d] = 0;
```

Comment:
Why do we haver duplicated code from `main`?

Reply:
Agree - main should only call this function, not compute ranks and loop over sizes

Reply:
replaced with `DimBlockPartitioner`